ledger-core/conveyor/conveyor_semaphore.go

```diff
@@ -26,6 +23,12 @@ func NewParallelProcessingLimiter(maxTasks int) ParallelProcessingLimiter {
 	}
 }
 
+func NewParallelProcessingLimiterByCPU() ParallelProcessingLimiter {
+	maxTasks := runtime.NumCPU() - 2
+
```

Comment:
check for maxTasks <= 0 here
and remove the check from NewParallelProcessingLimiter

Reply:
done